src/ripple/rpc/tests/KeyGeneration.test.cpp

```diff
@@ -0,0 +1,174 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2015 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/basics/StringUtilities.h>
+#include <ripple/basics/TestSuite.h>
+#include <ripple/rpc/handlers/WalletPropose.h>
+#include <ripple/rpc/impl/KeypairForSignature.h>
+
+namespace ripple {
+
+namespace RPC {
+
+struct key_strings
+{
+    char const* account_id;
+    char const* master_key;
+    char const* master_seed;
+    char const* master_seed_hex;
+    char const* public_key;
+    char const* public_key_hex;
+    char const* secret_key_hex;
+};
+
+namespace common {
+static char const* passphrase = "REINDEER FLOTILLA";
```

Comment:
Anonymous namespaces and `static` functions are orthogonal.  The former provides a compiler-generated namespace unique to the translation unit, and the latter uses static rather than extern linkage.  Anonymous namespaces exist to allow private symbol names to be used as template parameters.

I'm used to compilers where `char const foo[] = "bar";` allocates and populates a buffer on the stack.  If g++ and clang are smarter than that, I'll change it.